src/will.rs

```diff
@@ -89,20 +89,19 @@ impl<'a> Will<'a> {
         topic_len + payload_len + prop_len
     }
 
-    /// Set the retained status of the will.
-    ///
-    /// # Args
-    /// * `retained` - Specifies the retained state of the will.
-    pub fn retained(&mut self, retained: Retain) {
-        self.retained = retained;
+    /// Specify the will as a retained message.
+    pub fn set_retained(mut self) -> Self {
```

Comment:
```suggestion
    pub fn retained(mut self) -> Self {
```
for builder pattern.